src/QDL/firehose.js

```diff
@@ -0,0 +1,468 @@
+import { xmlParser } from "./xmlParser"
+import { concatUint8Array, containsBytes, compareStringToBytes, sleep, fromUint8ArrayToNumber, readBlobAsBuffer } from "./utils"
+import * as gpt from "./gpt"
+import * as Sparse from "./sparse";
+
+
+class response {
+  constructor(resp=false, data=new Uint8Array(), error="", log=[]) {
+    this.resp   = resp;
+    this.data   = data;
+    this.error  = error;
+    this.log    = log;
+  }
+}
+
+
+class cfg {
+  constructor() {
+    this.ZLPAwareHost                     = 1;
+    this.SkipStorageInit                  = 0;
+    this.SkipWrite                        = 0;
+    this.MaxPayloadSizeToTargetInBytes    = 1048576;
+    this.MaxPayloadSizeFromTargetInBytes  = 4096;
+    this.MaxXMLSizeInBytes                = 4096;
+    this.bit64                            = true;
+    this.total_blocks                     = 0;
+    this.block_size                       = 0;
+    this.SECTOR_SIZE_IN_BYTES             = 4096;
+    this.MemoryName                       = "UFS";
+    this.maxlun                           = 6;
+  }
+}
+
+
+export class Firehose {
+  constructor(cdc) {
+    this.cdc  = cdc;
+    this.xml  = new xmlParser();
+    this.cfg  = new cfg();
+    this.luns = [];
+  }
+
+
+  getStatus(resp) {
+    if (resp.hasOwnProperty("value")) {
+      let value = resp["value"];
+      return (value === "ACK" || value === "true");
+    }
+    return true;
+  }
+
+
+  async xmlSend(data, wait=true) {
+    let dataToSend = new TextEncoder().encode(data).slice(0, this.cfg.MaxXMLSizeInBytes);
+    await this.cdc?.write(dataToSend, null, wait);
+
+    let rData   = new Uint8Array();
```

Comment:
this style is inconsistent with the rest of the repo
```suggestion
    let rData  = new Uint8Array();
```